src/Build/Collections/CopyOnWritePropertyDictionary.cs

```diff
@@ -299,7 +299,8 @@ public bool Equals(CopyOnWritePropertyDictionary<T> other)
         /// </summary>
         bool IDictionary<string, T>.ContainsKey(string key)
         {
-            return PropertyNames.Contains(key);
+            T discard;
+            return ((IDictionary<string, T>)this).TryGetValue(key, out discard);
```

Comment:
Did you consider

```c#
return _properties.ContainsKey(key);
```

? That seems a bit more straightforward, and avoids memory allocations for the unneeded `T`.

Reply:
I stepped through this to compare what's happening. Aside from the allocation this is better in other ways. the sheer decrease in branch depth is worth it from my perspective. 
the old solution went through tons of functions before eventually landing where the case insensitive lookup was happening; by going through the wrapped dictionary's containskey that path is way more direct. 

Good call